### PR TITLE
Fix thinblock stats map interators from going out of range

### PR DESCRIPTION
**Pull request by ptschip**

### CODE REVIEW

File: src/thinblock.cpp [hunk 1]
```diff
@@ -290,7 +290,10 @@ void CThinBlockStats::UpdateInBound(uint64_t nThinBlockSize, uint64_t nOriginalB
 
     // Delete any entries that are more than 24 hours old
     int64_t nTimeCutoff = GetTimeMillis() - 60*60*24*1000;
-    for (std::map<int64_t, std::pair<uint64_t, uint64_t> >::iterator mi = CThinBlockStats::mapThinBlocksInBound.begin(); mi != CThinBlockStats::mapThinBlocksInBound.end(); ++mi) {
+    std::map<int64_t, std::pair<uint64_t, uint64_t> >::iterator iter = CThinBlockStats::mapThinBlocksInBound.begin();
+    while (iter != CThinBlockStats::mapThinBlocksInBound.end())
+    {
+        std::map<int64_t, std::pair<uint64_t, uint64_t> >::iterator mi = iter++; // increment to avoid iterator becoming invalid
         if ((*mi).first < nTimeCutoff)
             CThinBlockStats::mapThinBlocksInBound.erase(mi);
```
Comment (ftrader) [inline src/thinblock.cpp:298]: erase() would return the next element or point to .end(), using `mi` in this construct seems somewhat unnecessary... see here for what I mean:

http://stackoverflow.com/questions/4645705/vector-erase-iterator/4645727#4645727

**Comment by ptschip** [comment]
@ftrader   thanks, much simpler now.

**Comment by ptschip** [comment]
@ftrader actually no, that doesn't work, erase() returns a void...I'll have to revert.  I think on C++11 it would work fine though.

**Comment by ftrader** [comment]
Sorry, you're correct, only works since C++11.

**Pull request merged**